src/openrct2/ride/Ride.cpp

```diff
@@ -1276,34 +1276,31 @@ void ride_remove_provisional_track_piece()
         return;
     }
 
-    int32_t x = _unkF440C5.x;
-    int32_t y = _unkF440C5.y;
-    int32_t z = _unkF440C5.z;
+    auto coords = _unkF440C5;
     if (ride->type == RIDE_TYPE_MAZE)
     {
         int32_t flags = GAME_COMMAND_FLAG_APPLY | GAME_COMMAND_FLAG_ALLOW_DURING_PAUSED | GAME_COMMAND_FLAG_NO_SPEND
             | GAME_COMMAND_FLAG_GHOST;
-        maze_set_track(x, y, z, flags, false, 0, rideIndex, GC_SET_MAZE_TRACK_FILL);
-        maze_set_track(x, y + 16, z, flags, false, 1, rideIndex, GC_SET_MAZE_TRACK_FILL);
-        maze_set_track(x + 16, y + 16, z, flags, false, 2, rideIndex, GC_SET_MAZE_TRACK_FILL);
-        maze_set_track(x + 16, y, z, flags, false, 3, rideIndex, GC_SET_MAZE_TRACK_FILL);
+        maze_set_track({ coords, 0 }, flags, false, rideIndex, GC_SET_MAZE_TRACK_FILL);
+        maze_set_track({ coords + CoordsXYZ{ 0, 16, 0 }, 1 }, flags, false, rideIndex, GC_SET_MAZE_TRACK_FILL);
+        maze_set_track({ coords + CoordsXYZ{ 16, 16, 0 }, 2 }, flags, false, rideIndex, GC_SET_MAZE_TRACK_FILL);
+        maze_set_track({ coords + CoordsXYZ{ 16, 0, 0 }, 3 }, flags, false, rideIndex, GC_SET_MAZE_TRACK_FILL);
     }
     else
     {
-        int32_t direction = _unkF440C5.direction;
-        if (!(direction & 4))
+        if (!(coords.direction & 4))
```

Comment:
_unkF440C5 should not have been converted to a CoordsXYZD. Can you change it to a CoordsXYZ with the direction component separate.

Reply:
I haven't converted it, it was already a CoordsXYZD.

Reply:
How do I get the direction if _unkFF440C5 doesn't have it?

Reply:
You would need a separate variable just for direction. Pretty sure this is a global just add another global for now.